services/vision/vision.go

```diff
@@ -315,6 +325,72 @@ func (vm *vizModel) GetProperties(ctx context.Context, extra map[string]interfac
 	return &vm.properties, nil
 }
 
+func (vm *vizModel) CaptureAllFromCamera(
+	ctx context.Context,
+	cameraName string,
+	opt viscapture.CaptureOptions,
+	extra map[string]interface{},
+) (viscapture.VisCapture, error) {
+	ctx, span := trace.StartSpan(ctx, "service::vision::ClassificationsFromCamera::"+vm.Named.Name().String())
+	defer span.End()
+	cam, err := camera.FromRobot(vm.r, cameraName)
+	if err != nil {
+		return viscapture.VisCapture{}, errors.Wrapf(err, "could not find camera named %s", cameraName)
+	}
+	img, release, err := camera.ReadImage(ctx, cam)
+	if err != nil {
+		return viscapture.VisCapture{}, errors.Wrapf(err, "could not get image from %s", cameraName)
+	}
+	defer release()
+	var detections []objectdetection.Detection
+	if opt.ReturnDetections {
+		if !vm.properties.DetectionSupported {
```

Comment:
@bhaney I changed those tests to use the new `vm.properites`. Is that ok?

Reply:
Yes 👍